internal/services/resource/management_lock_resource.go

```diff
@@ -100,6 +100,19 @@ func resourceManagementLockCreate(d *pluginsdk.ResourceData, meta interface{}) e
 		return fmt.Errorf("creating %s: %+v", id, err)
 	}
 
+	stateConf := &pluginsdk.StateChangeConf{
+		Target: []string{
+			"OK",
+		},
+		Refresh:                   managementLockStateRefreshFunc(ctx, client, id),
+		MinTimeout:                10 * time.Second,
+		ContinuousTargetOccurence: 12,
+		Timeout:                   d.Timeout(pluginsdk.TimeoutUpdate),
+	}
+	if _, err := stateConf.WaitForStateContext(ctx); err != nil {
+		return fmt.Errorf("waiting for %s to finish create replication", id)
+	}
```

Comment:
We should be getting the deadline which will give us the remaining time for the create and not using the update timeout which was removed above?
```suggestion
	deadline, ok := ctx.Deadline()
	if !ok {
		return fmt.Errorf("internal-error: context was missing a deadline")
	}
	stateConf := &pluginsdk.StateChangeConf{
		Target: []string{
			"OK",
		},
		Refresh:                   managementLockStateRefreshFunc(ctx, client, id),
		MinTimeout:                10 * time.Second,
		ContinuousTargetOccurence: 12,
		Timeout:                   time.Until(deadline),
	}
	if _, err := stateConf.WaitForStateContext(ctx); err != nil {
		return fmt.Errorf("waiting for %s to finish create replication", id)
	}
```